src/mca/pmdl/mpich/pmdl_mpich_component.c

```diff
@@ -0,0 +1,96 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2004-2008 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2005 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2015      Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2016-2020 Intel, Inc.  All rights reserved.
+ * Copyright (c) 2021      Nanook Consulting.  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ *
+ */
+
+#include "src/include/pmix_config.h"
+#include "include/pmix_common.h"
+
+#include "pmdl_mpich.h"
+#include "src/mca/pmdl/pmdl.h"
+#include "src/util/argv.h"
+
+static pmix_status_t component_register(void);
+static pmix_status_t component_query(pmix_mca_base_module_t **module, int *priority);
+
+/*
+ * Instantiate the public struct with all of our public information
+ * and pointers to our public functions in it
+ */
+pmix_pmdl_mpich_component_t mca_pmdl_mpich_component = {
+    .super = {
+        .base = {
+            PMIX_PMDL_BASE_VERSION_1_0_0,
+
+            /* Component name and version */
+            .pmix_mca_component_name = "mpich",
+            PMIX_MCA_BASE_MAKE_VERSION(component,
+                                       PMIX_MAJOR_VERSION,
+                                       PMIX_MINOR_VERSION,
+                                       PMIX_RELEASE_VERSION),
+
+            /* Component open and close functions */
+            .pmix_mca_register_component_params = component_register,
+            .pmix_mca_query_component = component_query,
+        },
+        .data = {
+            /* The component is checkpoint ready */
+            PMIX_MCA_BASE_METADATA_PARAM_CHECKPOINT
+        }
+    },
+    .include = NULL,
+    .exclude = NULL
+};
+
+static pmix_status_t component_register(void)
+{
+    pmix_mca_base_component_t *component = &mca_pmdl_mpich_component.super.base;
+
+    mca_pmdl_mpich_component.incparms = "MPIR_*";
```

Comment:
I think it is safe to use "MPIR_CVAR" for all MPICH env vars.

Reply:
Good point. Question: Hydra has options about forwarding envars. It wasn't quite clear to me if those options apply to these envars, or were aimed at general envars other than the MPICH-specific ones. Any insights would be appreciated.